src/main/java/uk/gov/pay/connector/gateway/stripe/handler/StripeRefundHandler.java

```diff
@@ -39,9 +44,24 @@ public StripeRefundHandler(
     }
 
     public GatewayRefundResponse refund(RefundGatewayRequest request) {
+        StripeRefund refundResponse;
         try {
-            StripeRefund refundResponse = refundCharge(request);
-            transferFromConnectAccount(request);
+            if (usePaymentIntent(request)) {
+                StripePaymentIntent stripePaymentIntent = getPaymentIntent(request);
+                Optional<StripeCharge> maybeStripeCharge = stripePaymentIntent.getCharge();
+                if (maybeStripeCharge.isEmpty()) {
+                    throw new GatewayException.GenericGatewayException(
+                            String.format("Stripe charge not found for payment intent id %s", request.getTransactionId())
+                    );
+                    
+                }
+                StripeCharge stripeCharge = maybeStripeCharge.get();
+                refundResponse = refundCharge(request, stripeCharge.getId());
+                transferFromConnectAccount(request, stripeCharge.getId());
+            } else {
+                refundResponse = refundCharge(request, null);
+                transferFromConnectAccount(request, null);
```

Comment:
Can we not pass transactionId here rather than derive it in `StripeTransferInRequest of(RefundGatewayRequest re... `?  
```suggestion
                transferFromConnectAccount(request, request.getTransactionId());
```

Reply:
yeah, might be a little bit nicer, but this will all be cleaned up post release anyway